contracts/Cash.sol

```diff
@@ -50,4 +56,48 @@ contract Cash is ERC20Burnable, Operator {
     {
         super.burnFrom(account, amount);
     }
+    
+    function _transfer(address sender, address recipient, uint256 amount) 
+        internal 
+        override 
+    {
+        require(sender != address(0), "ERC20: transfer from the zero address");
+        require(recipient != address(0), "ERC20: transfer to the zero address");
+
+        if (amount > 0) {
+            _beforeTokenTransfer(sender, recipient, amount);
+            _balances[sender] = _balances[sender].sub(amount, "ERC20: transfer amount exceeds balance");
+
+            if(feeChecker.isTransferTaxed(sender, recipient)) {
+                uint256 feeAmount = IFeeChecker(feeChecker).calculateFeeAmount(sender, recipient, amount);
+                uint256 transferToAmount = amount.sub(feeAmount);
+
+                require(transferToAmount.add(feeAmount) == amount, "transferToAmount + feeAmount != amount");
+
+                _balances[recipient] = _balances[recipient].add(transferToAmount);
+                emit Transfer(sender, recipient, transferToAmount);
+
+                if(feeAmount > 0) {
+                    _balances[feeDistributor] = _balances[feeDistributor].add(feeAmount);
+                    emit Transfer(sender, feeDistributor, feeAmount);
+                    IFeeDistributor(feeDistributor).addFee(feeAmount);
```

Comment:
i don't think that we want to waste gas during tax collection to do the fee distribution and then afterwards burn 50% (bounty 3.3) and only 50% should be transfered to boardroom (bounty 3.4)

so just send half the feeAmount to the feeDistributor and burn the rest.

Reply:
Having the fee distributor handle sending the tokens to the other contracts/burning them makes it possible to change what we do with the fee, which is why I send tokens to the fee distributor contract instead of sending to boardroom + burning.